lib/msf/ui/console/command_dispatcher/db.rb

```diff
@@ -101,42 +101,64 @@ def active?
     true
   end
 
+  @@workspace_opts = Rex::Parser::Arguments.new(
+    [ '-h', '--help' ] => [ false, 'Help banner.'],
+    [ '-a', '--add' ] => [ true, 'Add a workspace.', '<name>'],
+    [ '-d', '--delete' ] => [ true, 'Delete a workspace.', '<name>'],
+    [ '-D', '--delete-all' ] => [ false, 'Delete all workspaces.'],
+    [ '-r', '--rename' ] => [ true, 'Rename a workspace.', '<old> <new>'],
+    [ '-l', '--list' ] => [ false, 'List workspaces.'],
+    [ '-v', '--list-verbose' ] => [ false, 'List workspaces verbosely.'],
+    [ '-S', '--search' ] => [ true, 'Search for a workspace.', '<name>']
+  )
+
   def cmd_workspace_help
     print_line "Usage:"
-    print_line "    workspace                  List workspaces"
-    print_line "    workspace -v               List workspaces verbosely"
-    print_line "    workspace [name]           Switch workspace"
-    print_line "    workspace -a [name] ...    Add workspace(s)"
-    print_line "    workspace -d [name] ...    Delete workspace(s)"
-    print_line "    workspace -D               Delete all workspaces"
-    print_line "    workspace -r <old> <new>   Rename workspace"
-    print_line "    workspace -h               Show this help information"
-    print_line
+    print_line "    workspace          List workspaces"
+    print_line "    workspace [name]   Switch workspace"
+    print_line @@workspace_opts.usage
   end
 
   def cmd_workspace(*args)
     return unless active?
+
+    adding = false
+    deleting = false
+    delete_all = false
+    renaming = false
+    list = false
```

Comment:
Not a blocker, potentially a single state is what we want here? that way we can't be adding/deleteing/renaming all at the same time:
```
state = nil
```

And then we could guard clause the argument parsing:
```
      when '-d','--del'
          return cmd_workspace_help if state != nil
          state = :delete
          # ....
```

Eventually:
```
if state == :add and names
```